src/main/scala/com/mozilla/telemetry/streaming/ErrorAggregator.scala

```diff
@@ -170,48 +170,53 @@ object ErrorAggregator {
       .withColumn("window_end", $"window.end")
       .drop("window")
   }
-  private def buildDimensions(meta: Meta): Row = {
-    val dimensions = new RowBuilder(dimensionsSchema)
-    dimensions("timestamp") = Some(meta.normalizedTimestamp())
-    dimensions("submission_date") = Some(new Date(meta.normalizedTimestamp().getTime))
-    dimensions("channel") = Some(meta.normalizedChannel)
-    dimensions("version") = meta.`environment.build`.flatMap(_.version)
-    dimensions("build_id") = meta.`environment.build`.flatMap(_.buildId)
-    dimensions("application") = Some(meta.appName)
-    dimensions("os_name") = meta.`environment.system`.map(_.os.name)
-    dimensions("os_version") = meta.`environment.system`.map(_.os.version)
-    dimensions("architecture") = meta.`environment.build`.flatMap(_.architecture)
-    dimensions("country") = Some(meta.geoCountry)
-    meta.experiment.foreach(experiment =>{
-      dimensions("experiment_id") = Some(experiment._1)
-      dimensions("experiment_branch") = Some(experiment._2)
-    })
-    dimensions("e10s_enabled") = meta.`environment.settings`.flatMap(_.e10sEnabled)
-    dimensions("e10s_cohort") = meta.`environment.settings`.flatMap(_.e10sCohort)
-    dimensions("gfx_compositor") = for {
-      system <- meta.`environment.system`
-      gfx <- system.gfx
-      features <- gfx.features
-      compositor <- features.compositor
-    } yield compositor
-    dimensions("quantum_ready") = meta.isQuantumReady
-    dimensions.build
+
+  private def buildDimensions(meta: Meta): Array[Row] = {
+
+    // add a null experiment_id and experiment_branch for each ping
+    val experiments = (meta.experiments :+ (None, None)).toSet.toArray
```

Comment:
I'm not 100% sure about adding a null experiment id/branch. If we do so we will not be able to tell which users were not enrolled in any experiment. This is not a blocker for the review but it would be good to have chat about it.